homeassistant/components/smarttub/config_flow.py

```diff
@@ -24,6 +25,12 @@ class SmartTubConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
     VERSION = 1
     CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
 
+    def __init__(self) -> None:
+        """Instantiate config flow."""
+        super().__init__()
+        self._reauth_input = None
+        self._reauth_entry = None
+
     async def async_step_user(self, user_input=None):
         """Handle a flow initiated by the user."""
         errors = {}
```

Comment:
If you reverse the conditional to `if user_input is not None:` you can fall though and show the form again when it fails.

Also build the database schema with defaults 

DATA_SCHEMA = vol.Schema(
    {vol.Required(CONF_EMAIL, default=self._reauth_input.get(CONF_EMAIL)): str, vol.Required(CONF_PASSWORD): str}
)

Then they can still change the email address if they have updated their email